geode-core/src/main/java/org/apache/geode/distributed/internal/ClusterDistributionManager.java

```diff
@@ -3231,13 +3231,13 @@ ExecutorService getThrottledSerialExecutor(
       ExecutorService executor = getSerialExecutor(sender);
 
       // Get the total serial queue size.
-      int totalSerialQueueMemSize = stats.getSerialQueueBytes();
+      int totalSerialQueueMemSize = stats.getInternalSerialQueueBytes();
 
       // for tcp socket reader threads, this code throttles the thread
       // to keep the sender-side from overwhelming the receiver.
       // UDP readers are throttled in the FC protocol, which queries
       // the queue to see if it should throttle
-      if (stats.getSerialQueueBytes() > TOTAL_SERIAL_QUEUE_THROTTLE
+      if (stats.getInternalSerialQueueBytes() > TOTAL_SERIAL_QUEUE_THROTTLE
```

Comment:
Should this use "totalSerialQueueMemSize" instead of calling getInternalSerialQueueBytes() again? I also don't understand why we keep using totalSerialQueueMemSize in the loop to calculate throttlePercent (since the stat is expected to be changing). I think you plan another PR to clean up this method so it is okay with me if you do not change this method on this PR.

Reply:
Oh yeah, why are we getting this value so many times. Seems we should always fetch it or only fetch it once.

Reply:
This will be fixed in [GEODE-6943 ](https://issues.apache.org/jira/browse/GEODE-6943 )